algosdk/v2client/algod.py

```diff
@@ -113,6 +113,11 @@ def algod_request(
             try:
                 return json.load(resp)
             except Exception as e:
+                # Some algod responses currently return a 200 OK
+                # but have an empty response.
+                # Do not return an error, and just return an empty response.
+                if resp.status == 200:
```

Comment:
nit. If at all possible, prefer to explicitly anticipate the "happy path" without catching an exception. I.e., even though its a bit more convoluted to explicitly check for an empty response along with a `200` then to try/catch as here, it'll be a bit more robust if there's a bug whereby algod returns a 200 errantly and has an unparseable response that is alerting of an error. Alternately, if `json.load()` gives a special error when an empty body is provided, you can selectively catch that _specific_ `Exception`-type instead of the generic type.

Reply:
Another approach would be to have an explict "white list" of acceptable endpoints that we know shouldn't return a body... But that seems like a maintenance hassle.

Reply:
A little less annoying would be to provide a new parameter such as `empty_response` defaulting to `False`.

Reply:
> Alternately, if json.load() gives a special error when an empty body is provided, you can selectively catch that specific Exception-type instead of the generic type.

Unfortunately, I don't think there is a separate exception for empty bodies vs bad bodies (parsing error).

> A little less annoying would be to provide a new parameter such as empty_response defaulting to False.

I like this approach at the moment - Ideally the server should return a different response (`204`) if there is an empty body, but that seems like a bigger hill to climb.

Reply:
I jumped the gun here, it looks like there are two ways an empty response is returned from algod:

1. We return a `NoContent` response in the v2 endpoints: https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/v2/handlers.go#L1114
2. We encode a `nil` into the response in the common endpoints: https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/common/handlers.go#L92

For 1, the server returns a length 0 response which causes an exception when parsing in Python, but for 2, the response is of length 5 for some reason and it parses to a `None` object in Python (without raising an exception). To handle case 1, it seems sufficient to just check the response length and ensure that it is 0.